contracts/Boardroomv2.sol

```diff
@@ -188,12 +205,45 @@ contract Boardroomv2 is ShareWrapper, ContractGuard, Epoch {
         claimReward();
     }
 
-    function claimReward() public updateReward(msg.sender) {
-        uint256 reward = directors[msg.sender].rewardEarned;
-        if (reward > 0) {
-            directors[msg.sender].rewardEarned = 0;
-            cash.safeTransfer(msg.sender, reward);
-            emit RewardPaid(msg.sender, reward);
+    function claimReward(uint256 amount) public updateReward(msg.sender) {
+        uint256 totalEarned = earned(msg.sender);
+        require(amount <= totalEarned, 'Amount cannot be larger than total claimable rewards');
+
+        for (uint i = directors[msg.sender].startEpoch; amount > 0; i++) {
+            uint256 claimable = calculateClaimableRewardsForEpoch(msg.sender, i);
+
+            if (amount > claimable) {
+                directors[msg.sender].rewardEarned[i] = 0;
+                directors[msg.sender].startEpoch = i.add(1);
+                amount = amount.sub(claimable);
+            } else {
+                removeRewardsForEpoch(msg.sender, amount, i);
+                amount = 0;
+            }
+        }
+
+        // In this case, startEpoch will be calculated again for the next stake
+        if (amount == totalEarned) {
+            directors[msg.sender].startEpoch = 0;
+        }
+
+        cash.safeTransfer(msg.sender, amount);
+        emit RewardPaid(msg.sender, reward);
```

Comment:
@chainlito This should be 'amount', not 'reward' right ?

Reply:
Oh, yes, you're right. Fixed it in 24b7c70a62a71825b98efa8a06325caa78ace57e